server/authorization_test.go

```diff
@@ -344,6 +344,75 @@ func TestEnvRequestCheck(t *testing.T) {
 	}
 }
 
+func TestBasePathStripping(t *testing.T) {
```

Comment:
We should probably test the converse as well - that the path should not be stripped for non-proxy type.

Reply:
Added tests